### PR TITLE
Fix `fission check` command doesnt work for namespace other than `fission`

### LINKED ISSUE #2924: "fission check" does not work outside of "fission" namespace
**Fission/Kubernetes version**

<pre>
$ fission version
client:
  fission/core:
    BuildDate: "2024-01-14T15:43:35Z"
    GitCommit: 7e8d5dd7
    Version: v1.20.1
server:
  fission/core:
    BuildDate: "2024-01-14T15:43:35Z"
    GitCommit: 7e8d5dd7
    Version: v1.20.1
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.17", GitCommit:"22a9682c8fe855c321be75c5faacde343f909b04", GitTreeState:"clean", BuildDate:"2023-08-23T23:44:35Z", GoVersion:"go1.20.7", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.17", GitCommit:"22a9682c8fe855c321be75c5faacde343f909b04", GitTreeState:"clean", BuildDate:"2023-08-23T23:37:25Z", GoVersion:"go1.20.7", Compiler:"gc", Platform:"linux/amd64"}
</pre>

**Kubernetes platform (e.g. Google Kubernetes Engine)**

Standard Kubernetes installed manually.

**Describe the bug**
When installing fission in the `fission` namespace, everything works. But when installing it in another namespace, the `fission check` command fails event though I can create and test functions properly:
```
$ fission check
fission-services
--------------------
× failed to get executor deployment status
× failed to get router deployment status
× failed to get storagesvc deployment status
Error: Error found: failed to get webhook deployment status
× failed to get webhook deployment status

fission-version
--------------------
√ fission is up-to-date
```

Event though the deployments are ok:
```
$ kubectl get deployments
NAME             READY   UP-TO-DATE   AVAILABLE   AGE
buildermgr       1/1     1            1           4m16s
executor         1/1     1            1           4m16s
kubewatcher      1/1     1            1           4m16s
mqtrigger-keda   1/1     1            1           4m16s
router           1/1     1            1           4m16s
storagesvc       1/1     1            1           4m16s
timer            1/1     1            1           4m16s
webhook          1/1     1            1           4m16s
```

Using `fission check -n` does not fix the issue.

**To Reproduce**

Install Fission 1.20.1 with this values.yaml:
```yaml
builderNamespace: mynamespace
functionNamespace: mynamespace
defaultNamespace: mynamespace
```

**Expected result**
Fission should be installed and `fission check` should return that everything is okay.

**Actual result**
As shown above, the deployments are OK but `fission check` fails.

### PR DESCRIPTION
**Pull request by soharab-ic**
## Description

`fission check` command does not work outside of "fission" namespace. If user provides namespace then use it for running `fission check` command otherwise use `fission` as default namespace.

## Which issue(s) this PR fixes:

Fixes #2924

## Testing

- Fission installed in `fission` namespace.
```
$ fission check
fission-services
--------------------
√ executor is running fine
√ router is running fine
√ storagesvc is running fine
√ webhook is running fine

fission-version
--------------------
√ fission is up-to-date
```
- Fission installed in `check` namespace.
```
$ fission check
fission-services
--------------------
× failed to get executor deployment status
× failed to get router deployment status
× failed to get storagesvc deployment status
Error: Error found: failed to get webhook deployment status
× failed to get webhook deployment status

fission-version
--------------------
√ fission is up-to-date
```
```
$ fission check --namespace check
fission-services
--------------------
√ executor is running fine
√ router is running fine
√ storagesvc is running fine
√ webhook is running fine

fission-version
--------------------
√ fission is up-to-date
```

## Checklist:

- [ ] I ran tests as well as code linting locally to verify my changes. 
- [x] I have done manual verification of my changes, changes working as expected.
- [ ] I have added new tests to cover my changes.
- [ ] My changes follow contributing guidelines of Fission.
- [x] I have signed all of my commits.

### CODE REVIEW

File: pkg/fission-cli/cmd/check/check.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ import (
 	"github.com/fission/fission/pkg/fission-cli/cmd"
 	flagkey "github.com/fission/fission/pkg/fission-cli/flag/key"
 	"github.com/fission/fission/pkg/healthcheck"
+	"github.com/pkg/errors"
```
Comment (sanketsudake) [inline pkg/fission-cli/cmd/check/check.go:21]: Please avoid using pkg/errors. Use `errors` or `fmt.Wrapf` instead.

File: pkg/fission-cli/cmd/check/check.go [hunk 2]
```diff
@@ -32,13 +33,18 @@ func (opts *CheckSubCommand) do(input cli.Input) error {
 
 	checks := []healthcheck.CategoryID{}
 
+	userProvidedNS, _, err := opts.GetResourceNamespace(input, flagkey.NamespaceFunction)
```
Comment (sanketsudake) [inline pkg/fission-cli/cmd/check/check.go:36]: I guess `flagkey.Namespace` is more appropriate.

**Review by sanketsudake — CHANGES_REQUESTED** [review]

File: pkg/fission-cli/cmd/check/check.go (hunk 1 above)
Comment (soharab-ic) [reply pkg/fission-cli/cmd/check/check.go:21]: `errors` standard pkg does not have any formatting method.
`fmt.Wrapf` is not available in standard `fmt` pkg so i will use `fmt.Errorf`.
Is this okay?

Comment (sanketsudake) [reply pkg/fission-cli/cmd/check/check.go:21]: Yes sorry meant Errorf. Basically consider using standard go packages.

Comment (soharab-ic) [reply pkg/fission-cli/cmd/check/check.go]: Done

**Review by sanketsudake — APPROVED** [review]

**Pull request merged**